### PR TITLE
[chore](compile) using PCH for compilation acceleration under clang

### PR DESCRIPTION
**Pull request by deadlinefen**
# Proposed changes

Using PCH for compilation acceleration under clang

## Checklist(Required)

* [ ] Rename the macro definition `RETURN_NOT_OK` to `OLAP_RETURN_NOT_OK` under `be/src/olap/olap_define.h`.
* [ ] Possible error part: Template instantiation in `be/src/util/string_parser.cpp` moved to `be/src/util/string_parser.hpp`

### CODE REVIEW

File: be/src/olap/cumulative_compaction.cpp [hunk 1]
```diff
@@ -117,7 +117,7 @@ Status CumulativeCompaction::pick_rowsets_to_compact() {
     // candidate_rowsets may not be continuous
     // So we need to choose the longest continuous path from it.
     std::vector<Version> missing_versions;
-    RETURN_NOT_OK(find_longest_consecutive_version(&candidate_rowsets, &missing_versions));
+    OLAP_RETURN_NOT_OK(find_longest_consecutive_version(&candidate_rowsets, &missing_versions));
```
Comment (yiguolei) [inline be/src/olap/cumulative_compaction.cpp:120]: Do not add this new macro. If RETURN_NOT_OK is conflict with arrow's macro definition. You could remove this macro "RETURN_NOT_OK" totally and use RETURN_IF_ERROR instead. They are same.

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.24 seconds
 stream load tsv:          430 seconds loaded 74807831229 Bytes, about 165 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          62 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230506041559_clickbench_pr_139092.html

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by deadlinefen** [comment]
run buildall

**Comment by deadlinefen** [comment]
run buildall

**Comment by deadlinefen** [comment]
run buildall

**Comment by deadlinefen** [comment]
run buildall

**Comment by deadlinefen** [comment]
run buildall

**Comment by deadlinefen** [comment]
run buildall

**Comment by deadlinefen** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**